ultranest/popstepsampler.py

```diff
@@ -536,8 +536,219 @@ def __next__(
+        else:
+            self.slice_limit=lambda tleft,tright:tleft,tright
+
+        
+        
+
+    def __str__(self):
+        """Return string representation."""
+        return 'PopulationEllipticalSliceSampler(popsize=%d, nsteps=%d, generate_direction=%s, scale=%.g)' % (
+            self.popsize, self.nsteps, self.generate_direction, self.scale)
+
+    def region_changed(self, Ls, region):
+        """Act upon region changed. Currently unused."""
+        pass
+
+    
+    
+    def __next__(
+        self, region, Lmin, us, Ls, transform, loglike, ndraw=10,
+        plot=False, tregion=None, log=False
+    ):
+        """Sample a new live point.
+
+        Parameters
+        ----------
+        region: MLFriends object
+            Region
+        Lmin: float
+            current log-likelihood threshold
+        us: np.array((nlive, ndim))
+            live points
+        Ls: np.array(nlive)
+            loglikelihoods live points
+        transform: function
+            prior transform function
+        loglike: function
+            loglikelihood function
+        ndraw: int
+            not used
+        plot: bool
+            not used
+        tregion: bool
+            not used
+        log: bool
+            not used
+
+        Returns
+        -------
+        u: np.array(ndim) or None
+            new point coordinates (None if not yet available)
+        p: np.array(nparams) or None
+            new point transformed coordinates (None if not yet available)
+        L: float or None
+            new point likelihood (None if not yet available)
+        nc: int
+
+        """
+        nlive, ndim = us.shape
+        
+        assert nlive>=self.popsize, "The number of live points should be greater than the population size" 
+        # fill if empty:
+        if len(self.prepared_samples) == 0:
+            # choose live points
+            ilive = np.random.choice(nlive, size=self.popsize, replace=False)
+            allu = np.array(us[ilive,:])
+            allp = np.zeros((self.popsize, ndim))
+            allL = np.array(Ls[ilive])
+            nc = 0
+            n_throws=0
+                                       
+                
+            
+            interval_final=0. 
+            
+            for k in range(self.nsteps):
+                # Defining scale jitter
+                factor_scale=self.scale_jitter_func()
+                # Defining slice direction
+                v = self.generate_direction(allu, region,scale= 1.0)*self.scale*factor_scale
+                # limite of the slice based on the unit cube boundaries
+                tleft,tright= unitcube_line_intersection(allu, v)
+                # Defining bound of the slice
+                Theta_min_worker,Theta_max_worker = self.slice_limit(tleft,tright)
+                Theta_min,Theta_max=Theta_min_worker.copy(),Theta_max_worker.copy()
```

Comment:
require instead that any slice_limit function must return a copy. (in docstring if a parameter)

Reply:
I added the option for the slice_limit function in the param and I set the default to keep the intersection with the unit cube.